### PR TITLE
CB-17697 Get rid of PowerMockito

### PR DESCRIPTION
**Pull request by bergerdenes**
And won't allow it ever again.
Migrated to JUnit 5 in the process.

### CODE REVIEW

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/SaltBootstrap.java [hunk 1]
```diff
@@ -48,16 +48,22 @@ public class SaltBootstrap implements OrchestratorBootstrap {
 
     private final BootstrapParams params;
 
+    private final SaltStates saltStates;
+
     private Set<Node> targets;
 
-    public SaltBootstrap(SaltConnector sc, Collection<SaltConnector> saltConnectors, List<GatewayConfig> allGatewayConfigs, Set<Node> targets,
+    /*
+        Intentionally Package-private to be able to verify construction later
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/SaltBootstrap.java:56]: could you elaborate?

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java [hunk 2]
```diff
@@ -53,6 +54,7 @@
 import com.sequenceiq.cloudbreak.orchestrator.salt.domain.StateType;
 import com.sequenceiq.cloudbreak.service.Retry;
 
+@Service
 public class SaltStates {
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java:58]: I feel like this should be renamed to `SaltStateService` or similar if it is a service

File: orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestratorTest.java [hunk 3]
```diff
@@ -744,7 +644,7 @@ public void testNewInstanceIsPrimaryReplacement() throws Exception {
         HostAndRoleTarget replicaTarget = new HostAndRoleTarget("freeipa_replica", Set.of("newInstance.domain", "replica2.domain"));
         ArrayNode replicaRole = mapper.createArrayNode();
         replicaRole.add("freeipa_replica");
-        PowerMockito.when(SaltStates.getGrains(eq(saltConnector), eq(replicaTarget), eq("roles")))
+        lenient().when(saltStates.getGrains(eq(saltConnector), eq(replicaTarget), eq("roles")))
```
Comment (lacikaaa) [inline orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestratorTest.java:647]: why `lenient` is required? it's not a parametrized test

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java (hunk 2 above)
Comment (bergerdenes) [reply orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java:58]: I agree, I did not want to make this step as of now but if you also feel like this is the right naming, then I will change it.

File: orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestratorTest.java (hunk 3 above)
Comment (bergerdenes) [reply orchestrator-salt/src/test/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestratorTest.java:647]: the `getGrains` called 3 times within the `saltOrchestrator.installFreeIpa`
PowerMockito swallowed this, the new Mockito does not. I can add 3 different `when`s if needed.

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/SaltBootstrap.java (hunk 1 above)
Comment (bergerdenes) [reply orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/SaltBootstrap.java:56]: PowerMockito checked for the creation of the `SaltBootsrap class`
In regular Mockito, verifying of constructor call is not possible so to workaround this I introduced a Factory class that instantiates `SaltBootstrap` and now the factory method can be verified.

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java (hunk 2 above)
Comment (lacikaaa) [reply orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java:58]: thanks, I think it would be nice to have

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**